bpf/mb_recvmsg.bpf.c

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright © 2022 Merbridge Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "headers/helpers.h"
+#include "headers/maps.h"
+#include "headers/mesh.h"
+
+#if ENABLE_IPV4
+__section("cgroup/recvmsg4") int mb_recvmsg4(struct bpf_sock_addr *ctx)
+{
+#if MESH != ISTIO && MESH != KUMA
+    // only works on istio and kuma
+    return 1;
+#endif
+    if (bpf_htons(ctx->user_port) != DNS_CAPTURE_PORT) {
+        return 1;
+    }
+    if (!(is_port_listen_current_ns(ctx, ip_zero, OUT_REDIRECT_PORT) &&
+          is_port_listen_udp_current_ns(ctx, localhost, DNS_CAPTURE_PORT))) {
+        // printk("recv4 : not from pod");
```

Comment:
comment

Reply:
I would live it as it's present in the current merbridge codebase, I just updated the value of the log